### ISSUE TITLE
xunit template

### ISSUE THREAD

**Issue by Varorbc** [dotnet/sdk#45645]
With the release of xUnit v3, they also came out with new project template [packages](https://www.nuget.org/packages/xunit.v3.templates).  So, do we still need to maintain our current xUnit project templates?  Or will we be upgrading this xUnit to v3 as well?

**Comment by Evangelink** [comment]
@bradwilson What do you want to do with the xUnit templates of .NET 10? Shall we bump to v3?

cc @pavelhorak @smithkaylee @webreidi this would be a breaking change similar to what we want to do for MSTest with the same kind of implications for users not paying attention to updates (mixed mode issues etc).

**Comment by bradwilson** [comment]
I think timing wise it makes sense to bump up to v3. Ideally I would like
it modeled after our existing template (including the project comments, the
xunit.runner.json inclusion, etc.).
https://github.com/xunit/xunit/tree/main/src/xunit.v3.templates/templates/project

We'll continue to ship our separate v3 templates for the time being and
then make a decision if/when it makes sense to discontinue them.

**Comment by bradwilson** [comment]
I think timing wise it makes sense to bump up to v3. Ideally I would like it modeled after our existing template (including the project comments, the `xunit.runner.json inclusion`, etc.). https://github.com/xunit/xunit/tree/main/src/xunit.v3.templates/templates/project

We'll continue to ship our separate v3 templates for the time being and then make a decision if/when it makes sense to discontinue them.

**Comment by Youssef1313** [comment]
Pinging @baronfel @marcpopMSFT @dsplaisted @mckennabarlow

Should we bump xunit templates to xunit.v3 in .NET 10?

**Comment by baronfel** [comment]
It's a bit late for impactful changes to the templates - p7 has historically been the date so Tactics will likely want to vet this. 

I think it's entirely reasonable that the SDK ship with what is most up to date at point of release. Xunit 3 is the future and 2 is becoming more and more legacy every day. In addition, IIRC 2 will never be compatible with the MTP. So overall I'd support this change.

**Comment by Youssef1313** [comment]
@baronfel That's correct. xunit 2 isn't going to support MTP. The future is xunit.v3. I opened #50117. Hopefully it can be accepted at this time?

**Comment by Evangelink** [comment]
Wendy was against this change. Please let's have Mc Kenna in the loop

**Comment by Varorbc** [comment]
https://devblogs.microsoft.com/dotnet/dotnet-test-with-mtp/

Since the article states that MTP is the future, why is it not the default option in the template, but rather asking users to uncomment? This experience is not very good.

**Comment by bradwilson** [comment]
@Varorbc I think one problem right now is that `dotnet.config` for .NET 10 SDK should be in the _solution_ folder, so there's not much that can be done in the _project_ templates.

I would have no problem with the template shipping with `TestingPlatformDotnetTestSupport` set to `true` by default. There is still some discussion to be had about whether that would be respected in .NET 10 SDK `dotnet test`, and I'm pushing back on adoption of MTP v2 based on whether that upgrade becomes a breaking change for users because it forces them to upgrade to .NET 10 SDK, even for projects that otherwise wouldn't need it.

I started my thoughts on the issue here: https://github.com/xunit/xunit/issues/3375#issuecomment-3208898699
